src/components/noteBlock/NoteBlock.tsx

```diff
@@ -122,30 +124,48 @@ const NoteBlock: React.FC<NoteBlockProps> = props => {
      * NotePage::deleteBlockHandler
      * NoteBlock::onKeydownHandler (ArrowUp and ArrowDown)
      */
-    <div
-      className="noteblock"
-      ref={props.innerRef}
-      {...props.provided.dragHandleProps} // react-beautiful-dnd props
-      {...props.provided.draggableProps} // react-beautiful-dnd props
-    >
-      <div className="noteblock-handle"></div>
-      <ContentEditable
-        className="noteblock-text"
-        innerRef={mergedRef}
-        html={props.html}
-        tagName={props.tag}
-        onChange={onChangeHandler}
-        onKeyDown={onKeydownHandler}
-        disabled={!props.isEditMode}
-        onClick={() => {
-          props.setIsEditMode(true, () => {
-            noteBlockRef.current?.focus();
-            // TODO: This eol line is causing issues when using the mouse to position the cursor
-            // setEol(noteBlockRef.current);
-          });
-        }}
-      />
-    </div>
+    <ContextMenuTrigger id={props.id} holdToDisplay={1000}>
+      <div
+        className="noteblock"
+        ref={props.innerRef}
+        {...props.provided.dragHandleProps} // react-beautiful-dnd props
+        {...props.provided.draggableProps} // react-beautiful-dnd props
+      >
+        <div className="noteblock-handle"></div>
+
+        <ContentEditable
+          className="noteblock-text"
+          innerRef={mergedRef}
+          html={props.html}
+          tagName={props.tag}
+          onChange={onChangeHandler}
+          onKeyDown={onKeydownHandler}
+          disabled={!props.isEditMode}
+          onClick={() => {
+            props.setIsEditMode(true, () => {
+              noteBlockRef.current?.focus();
+              // TODO: This eol line is causing issues when using the mouse to position the cursor
+              // setEol(noteBlockRef.current);
+            });
+          }}
+        />
+
+        <ContextMenu id={props.id}>
+          <Menu vertical>
+            <MenuItem onClick={() => {}}>
+              <Menu.Item onClick={() => props.deleteBlock(props, noteBlockRef)}>
+                <Icon name="trash alternate" />
+                Delete Block
+              </Menu.Item>
+              <Menu.Item onClick={() => props.addBlock(props, noteBlockRef)}>
+                <Icon name="add" />
+                Add Block
+              </Menu.Item>
+            </MenuItem>
+          </Menu>
+        </ContextMenu>
+      </div>
+    </ContextMenuTrigger>
```

Comment:
similarly can abstract this into the common context menu component mentioned earlier